server/events.ts

```diff
@@ -0,0 +1,26 @@
+import EventEmitter from "events";
+import Client from "./client";
+import {NetworkWithIrcFramework} from "./models/network";
+import {MessageData, NotificationData} from "./plugins/irc-events/message";
+export interface ClientEvents {
+	message: (
+		irc: NetworkWithIrcFramework["irc"],
+		network: NetworkWithIrcFramework,
+		client: Client,
+		data: MessageData
+	) => void;
+	notification: (
+		irc: NetworkWithIrcFramework["irc"],
+		network: NetworkWithIrcFramework,
+		client: Client,
+		data: MessageData,
+		notification: NotificationData
+	) => void;
```

Comment:
The event structure is fine, but in my opinion I really don't want to expose the irc framework nor any other internal data structures.

Shouldn't we hide that all behind a public API Interface?

If you expose it like that our internals become public API and we can't change it any longer and are exposing the whole of the framework API which we then need to carry should we ever wrap that deeper for dome reason.

Our current API has the same problem and this might be a me thing, but it seems wrong layer wise.

Reply:
Agreed. I wasn't exactly sure what the best way to handle that would be (and I was just sticking with what current command events fire).

Maybe we should discuss what we're interested in the client exposing and I can stub out a client interface?